crates/ruff_linter/resources/test/fixtures/ruff/for_variable_used_after_block.py

```diff
@@ -0,0 +1,81 @@
+for global_var in []:
+    _ = global_var
+    pass
+
+# ❌
+_ = global_var
+
+def foo():
+    # For control var used outside block
+    for event in []:
+        _ = event
+        pass
+
+    # ❌
+    _ = event
+
+    for x in []:
+        _ = x
+        pass
+    
+    # Using the same control variable name in a different loop is ok
+    for x in []:
+        _ = x
+        pass
+
+    for y in []:
+        # ❌ x is used outside of the loop it was defined in (meant to use y)
+        if x == 5:
+            pass
+
+    # Assign a variable before the loop
+    room_id = 3
+    _ = room_id
+    # Use the same variable name in a loop
+    for room_id in []:
+        _ = room_id
+        pass
+
+    # ❌ After the loop is not ok because the value is probably not what you expect
+    _ = room_id
+
+    # ❌ Augmented assignment is not allowed because the value is probably not what you expect
+    room_id += 1
+
+    # Assigning again after the loop is ok
+    room_id = 5
+    room_id += 1
+    _ = room_id
+
```

Comment:
#### Control flow

> This here seems like a clear false positive but detecting it would be hard because it requires some form of control flow analysis

The [`pandas` example](https://github.com/pandas-dev/pandas/blob/0cdc6a48302ba1592b8825868de403ff9b0ea2a5/pandas/plotting/_matplotlib/boxplot.py#L520-L561) makes sense to allow if we can detect the context via control flow analysis. The [other `pandas` example](https://github.com/pandas-dev/pandas/blob/0cdc6a48302ba1592b8825868de403ff9b0ea2a5/pandas/io/common.py#L581-L599) makes sense for this as well since it has a `return` statement to guard it every leaking below.

But I think it's good that we catch [`bokeh`](https://github.com/bokeh/bokeh/blob/829b2a75c402d0d0abd7e37ff201fbdfd949d857/src/bokeh/embed/util.py#L266-L276) [examples](https://github.com/bokeh/bokeh/blob/829b2a75c402d0d0abd7e37ff201fbdfd949d857/src/bokeh/plotting/_figure.py#L476-L486) as sketchy. I don't see how we could use control flow analysis to see that they're fine?